### PR TITLE
[Pylint] Update linter iterable rule

### PR DESCRIPTION
**Pull request by l0lawrence**
list_ -> ItemPaged
ItemPaged - > list_

Addressses: https://github.com/Azure/azure-sdk-for-python/issues/26681

### CODE REVIEW

File: tools/pylint-extensions/azure-pylint-guidelines-checker/pylint_guidelines_checker.py [hunk 1]
```diff
@@ -893,28 +893,32 @@ def __init__(self, linter=None):
...
+                if node.parent.name.startswith("list"):
+                    paging_class = True
 
-                    if not paging_class:
-                        self.add_message(
-                            msgid="client-list-methods-use-paging", node=node.parent, confidence=None
-                        )
+                if (not paging_class and iterable_return) or (paging_class and not iterable_return):
+                    self.add_message(
+                        msgid="client-list-methods-use-paging", node=node.parent, confidence=None
```
Comment (kristapratico) [inline tools/pylint-extensions/azure-pylint-guidelines-checker/pylint_guidelines_checker.py:919]: Can we update the linting message to cover both cases? Or emit a different message in the new case?

File: tools/pylint-extensions/azure-pylint-guidelines-checker/pylint_guidelines_checker.py [hunk 2]
```diff
@@ -865,19 +865,19 @@ def visit_call(self, node):
 class ClientListMethodsUseCorePaging(BaseChecker):
     __implements__ = IAstroidChecker
 
-    name = "client-list-methods-use-paging"
+    name = "client-paging-methods-use-list"
     priority = -1
     msgs = {
         "C4733": (
-            "Operations that return collections should return a value that implements the Paging protocol. See details:"
+            "Operations that return collections should return a value that implements the Paging protocol and be prefixed with list_ or _list_. See details:"
```
Comment (kristapratico) [inline tools/pylint-extensions/azure-pylint-guidelines-checker/pylint_guidelines_checker.py:872]: What's the case for `_list_`?

Comment (l0lawrence) [reply tools/pylint-extensions/azure-pylint-guidelines-checker/pylint_guidelines_checker.py:872]: Case for _list was after talking to Johan, even if it is a private _list function we still want it to conform to the SDK rules of returning an iterable, or if a private function returns an iterable we want it to be _list prefixed

File: tools/pylint-extensions/azure-pylint-guidelines-checker/tests/test_pylint_custom_plugins.py [hunk 3]
```diff
@@ -1955,15 +1955,47 @@ async def list_thing2(self, **kwargs): #@
...
+            ),
+        ):
+            self.checker.visit_return(function_node_a.body[0])
+
+    def test_finds_return_AsyncItemPaged_not_list(self):
+        class_node, function_node_a = astroid.extract_node("""
+        from azure.core.async_paging import AsyncItemPaged
+        
+        class SomeClient(): #@
+            async def some_thing(self): #@
```
Comment (kristapratico) [inline tools/pylint-extensions/azure-pylint-guidelines-checker/tests/test_pylint_custom_plugins.py:1988]: sorry just noticed this, this method shouldn't use async

**Review by kristapratico — APPROVED** [review]

**Pull request merged**